src/components/fileStorage/uc/applicationInternal/fileStorageProvider.js

```diff
@@ -0,0 +1,94 @@
+const CryptoJS = require('crypto-js');
+const { Configuration } = require('@hpi-schul-cloud/commons');
+const mongoose = require('mongoose'); // ToDo: Remove, currently used to initialize a session
+
+const fileStorageProviderRepo = require('../../repo/fileStorageProvider.repo');
+
+const { facadeLocator } = require('../../../../utils/facadeLocator');
+
+const createCopyParamsToMarkAsBeingDeleted = (bucket, fileId) => {
+	return {
+		Bucket: bucket,
+		CopySource: `/${bucket}/${fileId}`,
+		Key: `expiring_${fileId}`,
+		MetadataDirective: 'REPLACE',
+		Metadata: {
+			expires: true,
+		},
+	};
+};
+
+const createDeleteParams = (bucket, fileIds) => {
+	return {
+		Bucket: bucket,
+		Delete: {
+			Objects: fileIds.map((fileId) => ({
+				Key: fileId,
+			})),
+		},
+	};
+};
+
+const storageBucketName = (schoolId) => `bucket-${schoolId}`;
+
+const chooseProvider = async (schoolId) => {
+	let storageProvider;
+	const session = await mongoose.startSession(); // ToDo: How to allign with new architecture
```

Comment:
Do we really really need the session, or can we avoid it? It tackle or mongoose scaling strategy. 

If we decrease with an await, we have allocated our bucket place. Then we can execute setStorageProvider after it.